File: 09-May Valid Perfect Square/Valid_Perfect_Square_1.cpp
class Solution {
public:
    bool isPerfectSquare(int num){
        if(num == 0 || num == 1 || num == 4 || num == 9)
                return true;
        int z = (num%2);
        long int i = 4;
        if (z == 1)
            i = 3;
        int k =(num/4+1);
        for (; i < k; i=i+2){           //skips even nos. for odd squares and vice versa
            if ((i*i) == num){
                return true;
            }
        }
        return false;
    }
};

File: 09-May Valid Perfect Square/Valid_Perfect_Square_1.java
/*
1       = 1
1+3     = 4
1+3+5   = 9
1+3+5+7 = 16
*/

class Solution {
    public boolean isPerfectSquare(int num) {
        int i = 1;
        while(num > 0){
            num -= i;
            i += 2;
            if(num == 0){ 
                return true;
            }
        }
        return false;
    }
}
